### PR TITLE
Update and improve Riemann solver code

### PR DESCRIPTION
**Pull request by ketch**
This PR is intended to improve the existing code without changing any functionality.

- Use "implicit none" in all solvers, and declare types of all variables
- Eliminate old-style fortran code that is deprecated, such as numbered loops

It is far from complete.  Before merging it, I also want to get all the tests and continuous integration working in PyClaw.

### CODE REVIEW

**Comment by ketch** [comment]
This PR is now ready for consideration to merge and I've requested reviews from @mandli and @rjleveque .  Most of the work was done by @carlosmunozmoncayo .  Here are more details:

- `implicit none` is now used in all solvers
- I initially planned to correct indentation and remove loop numbers in all solvers, but this has only been done in some solvers.  I think it makes sense to merge this PR and then complete those tasks in a separate future PR.
- A major enhancement added by @carlosmunozmoncayo is that now for PRs in `riemann`, the CI will automatically run tests in the other repositories.

There are two currently failing tests, but both of them fail prior to this PR:

- the PyClaw test for shallow water on the sphere
- the AMRClaw test for adjoint acoustics

**Review by mandli — COMMENTED** [review]
Mostly just some general stylistic comments.  The testing needs to be fixed and I would suggest that we change all mentions of `double precision` to `real(kind=8)`.  I was also thinking that we could trigger only specific tests from the other packages, including GeoClaw so that it take more time and is more directed.

File: .github/workflows/amrclaw_tests.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+name: Test Riemann solvers with AMRClaw tests
+
+on:
+  push:
+    branches: [ "implicit_none" ]
```
[the remark this answers is not in the document]
Comment (ketch) [reply .github/workflows/amrclaw_tests.yml:5]: Ah yes, that was just a hack to get it to test this branch for the moment.

File: collect_pytests_amrclaw.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#Collect Pytest tests for AMRClaw
```
[the remark this answers is not in the document]
Comment (mandli) [reply collect_pytests_amrclaw.sh:1]: Modifications were required to get the collection to work as you mentioned.  [This](https://github.com/clawpack/amrclaw/pull/279) is hopefully going to fix that.  Not sure what to do until then.

**Comment by ketch** [comment]
As discussed separately with @mandli , the issues he has raised here are pre-existing things that are best fixed in a future PR.  Therefore, if there are no objections I will merge this tomorrow.

**Comment by mandli** [comment]
👍

**Pull request merged**